templates/default/set1.json.erb

```diff
@@ -45,6 +45,10 @@ end
      "resultAlias": "<%= query['result_alias'] %>",
      "obj": <%= escape_json(query['obj']) %>,
      "attr": [<%= query['attr'].map {|t| escape_json(t.to_s) }.join(', ') %>]
+     <% unless query['typeNames'].nil? -%>
+     ,
+     "typeNames": [<%= query['typeNames'].map {|t| escape_json(t.to_s) }.join(', ') %>]
```

Comment:
maybe you could have a simpler code by inserting the line above "attr"
```
<% if query['typeNames'] %>
"typeNames": [<%= query['typeNames'].map {|t| escape_json(t.to_s) }.join(', ') %>],
<% end %>
```

this way you handle the comma nicely.

Of course a better way would to be simple loop query key/values and escape_json the values (handling the array case), this would avoid to add a line each time you want to add filed to configuration.